### PR TITLE
Added Anagrams in C, Go and Java

### LINKED ISSUE #1284: Adding Anagrams program in C, Java and Go

### PR DESCRIPTION
**Pull request by singh-shreya6**
Fixes #1284 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added Anagrams in C, Go and Java 

#### Languages Used:
- C
- Java
- Go

#### Files Added:
- Anagram.c
- Anagram.java
- Anagram.go

### CODE REVIEW

File: Anagram/Anagram.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    {
+        count1[str1[i] - 'a']++;
+        count2[str2[i] - 'a']++;
+    }
+    
+    // If the counts of characters are not equal,
+    // they are not anagrams
+    for(i = 0; i < 256; i++)
+    {
+        if(count1[i] != count2[i])
```
Comment (marwahmanbir) [inline Anagram/Anagram.c]: Space after ``if``.

File: Anagram/Anagram.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+/*
+    Anagrams are strings formed by rearranging the
+    letters of another string.
+    Example: LISTEN and SILENT, as they have the same
+    characters just appearing in another order.
+*/
+
+package main
+
```
Comment (marwahmanbir) [inline Anagram/Anagram.go]: Remove this extra line.

File: Anagram/Anagram.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        String str1 = sc.nextLine();
+
+        System.out.println("Enter 2nd string");
+        String str2 = sc.nextLine();
+
+        int count1[] = new int[256];
+        int count2[] = new int[256];
+
+        // if lengths are not equal, they cannot be anagrams
+        if(str1.length() != str2.length())
```
Comment (marwahmanbir) [inline Anagram/Anagram.java]: Space after ``if``.

File: Anagram/Anagram.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            System.out.println("The strings are not anagrams");
+            return;
+        }
+    
+        // Store the count of every character in string
+        for(int i = 0; i < str1.length(); i++)
```
Comment (marwahmanbir) [inline Anagram/Anagram.java]: Space after ``for``

File: Anagram/Anagram.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            count1[str1.charAt(i) - 'a']++;
+            count2[str2.charAt(i) - 'a']++;
+        }
+    
+        // If the counts of characters are not equal,
+        // they are not anagrams
+        for(int i = 0; i < 256; i++)
```
Comment (marwahmanbir) [inline Anagram/Anagram.java]: Space after ``for``.

File: Anagram/Anagram.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            if(count1[i] != count2[i])
```
Comment (marwahmanbir) [inline Anagram/Anagram.java]: Space after ``if``.

**Comment by singh-shreya6** [comment]
@marwahmanbir I have made the requested changes.
Please review.

File: Anagram/Anagram.java (hunk 6 above)
Comment (singh-shreya6) [reply Anagram/Anagram.java]: Done @marwahmanbir

**Comment by marwahmanbir** [comment]
Good work.

**Pull request merged**